core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -121,10 +125,77 @@ protected void doStop() throws Exception {
       }
     };
 
-    NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(server);
-    connector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty", metricTags));
-    connector.setPort(getConfiguration().getInt(RestConfig.PORT_CONFIG));
-    server.setConnectors(new Connector[]{connector});
+    // http
+    if (config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP) ||
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(server);
+      connector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty", metricTags));
+      connector.setPort(getConfiguration().getInt(RestConfig.PORT_CONFIG));
+      server.addConnector(connector);
+      log.info("Added http connector.");
+    }
+
+    // https
+    if (config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTPS) ||
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      // IMPORTANT: the key's CN, stored in the keystore, must match the FQDN. This is a Jetty requirement.
```

Comment:
In my test, Jetty doesn't support SubjectAltNames. Here's the exception:

    Exception in thread "main" javax.net.ssl.SSLPeerUnverifiedException: Host name 'localhost' does not match the certificate subject provided by the peer (CN=Alex Loddengaard, OU=Services, O=Confluent, L=Palo Alto, ST=California, C=US)
    	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.verifyHostname(SSLConnectionSocketFactory.java:465)
    	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:395)
    	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:353)
    	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:141)
    	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
    	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
    	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
    	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
    	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
    	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
    	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
    	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
    	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
    	at TestSchemaRegistryHttps.makeGetRequest(TestSchemaRegistryHttps.java:49)
    	at TestSchemaRegistryHttps.main(TestSchemaRegistryHttps.java:17)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:497)
    	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)

I generated the keystore with `-ext san=dns:localhost`.

I've had trouble finding this in the jetty docs to confirm.